resources/git-rebase.md

```diff
@@ -0,0 +1,81 @@
+
+Another reason to rebase is to head off any potential merge conflicts. A merge conflict occurs when your branch and another branch have code edits on the same lines. Git doesn’t know which of the edits should be kept, so it throws a conflict that a human (you!) has to resolve.
+
+## Merge conflicts sound terrible. How can I make sure I never have to deal with them?
+
+Alas, dealing with merge conflicts is a part of developer life. But they don’t have to be awful! To mitigate conflicts, commit your work often and rebase master when new code is added. If everyone on your team does this, you’ll keep your local branches up-to-date and lower the entire team's odds of having to fix a large merge conflict.
+
+Using the interactive mode when performing a rebase helps as well. Interactive mode can be used by adding the -i flag to your rebase command like so:
+
+ `git rebase -i <nameOfBranchToRebaseIn>`
+
+Using interactive mode when rebasing lets you see the commits that you're rebasing, and allows you to edit and squash commits.
+
+## What is git squash?
+Git squash allows you to take several commits and combine them into a single commit. Say you've finished a feature - styling a nav bar - and your git history looks like this:
+```
+3f0e9cdcda init commit
+9845111670 in progress
+cd362d2f52 still working
+300571e57c styled nav bar
+```
+
+To keep your git history clean and easy to read, you can combine all of these commits into a single commit and update your commit message. You can do this by rebasing your current branch into itself, and then squashing your commits together, like this:
+
+`git rebase -i <nameOfBranchYouAreOn>~4`
+
+The reason "4" is added to the end of the rebase message is because your branch currently has 4 commits that you want to make into a single commit. After you run this command, git will show you the commits you've selected:
+
+```
+pick 3f0e9cdcda init commit
+pick 9845111670 in progress
+pick cd362d2f52 still working
+pick 300571e57c styled nav bar
+```
+
+You can squash your commits by keeping the first commit as is, and selecting 'squash' for the other commits like so:
+
+```
+pick 3f0e9cdcda init commit
+s 9845111670 in progress
+s cd362d2f52 still working
+s 300571e57c styled nav bar
+```
+
+Save your choices:
+`cmd + s`, then `cmd+ w`
+
+Once you've done this, git will give you the option of editing the commit messages for all of the commits you've changed. You can edit your commit messages into a single commit that describes the work you've done on the branch. When you're finished editing the messages, save and quit your editor.
+
+At this point you can push your new, squashed changes. You'll likely need to force push, so rather than:
+`git push`
+
+you'll need:
+
+`git push -f`
```

Comment:
Just a heads up that depending on how @cgolinski approaches the `git fork` content, we may want to add some context here for multiple remotes.

Reply:
hmm, yes, that's a good point. All of our git content should definitely be in sync. I'm not sure how to note this in this branch right now...a TODO comment? @BrendanFDMoore, what do you think?

Reply:
Shouldn't be a concern right now, but we can talk about it in the next meeting